test/integration/gardenlet/seed/care/care_test.go

```diff
@@ -177,6 +177,33 @@ var _ = Describe("Seed Care controller tests", func() {
 				WithMessageSubstrings("All system components are healthy."),
 			))
 		})
+
+		It("should delete stale pods in all namespaces except kube-system", func() {
+			var (
+				pod1 = &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{GenerateName: "pod-", Namespace: testNamespace.Name},
+					Spec:       corev1.PodSpec{Containers: []corev1.Container{{Name: "foo-container", Image: "foo"}}},
+				}
+				pod2 = &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{GenerateName: "pod-", Namespace: metav1.NamespaceSystem},
+					Spec:       corev1.PodSpec{Containers: []corev1.Container{{Name: "foo-container", Image: "foo"}}},
+				}
+			)
+
+			Expect(testClient.Create(ctx, pod1)).To(Succeed())
+			Expect(testClient.Create(ctx, pod2)).To(Succeed())
+
+			pod1.Status.Reason = "Evicted"
+			Expect(testClient.Status().Update(ctx, pod1)).To(Succeed())
+
+			Eventually(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(pod1), &corev1.Pod{})
+			}).Should(BeNotFoundError())
+
+			Consistently(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(pod2), &corev1.Pod{})
+			}).Should(Succeed())
+		})
```

Comment:
Maybe I am missing something, but this test does not look complete from my point of view. I would have expected to have `pod2` also being stale otherwise it would never be evicted anyway. Furthermore, there is no call to delete stale pod. Is that somehow happening magically?

Reply:
You're right, I was missing to also update the status of `pod2` to `Evicted`. This way, we ensure that stale pods in `kube-system` namespaces are ignored.

> Furthermore, there is no call to delete stale pod. Is that somehow happening magically?

This is an integration test - the controller running in the background is performing the deletion. This is actually what we want to test, i.e., whether the controller behaves as expected :)